src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/ApplicationServices/WindowsFormsApplicationBase.vb

```
@@ -829,6 +914,142 @@ Namespace Microsoft.VisualBasic.ApplicationServices
+            Permissions.AddPermission(New SecurityPermission(SecurityPermissionFlag.UnmanagedCode))
+            Permissions.Assert()
+
+            Dim Guid As Guid = GetTypeLibGuidForAssembly(Entry)
+            If Guid = Nothing Then
+                Return Entry.ManifestModule.ModuleVersionId.ToString
+            End If
+            Dim Version As String = Entry.GetName.Version.ToString
+            Dim VersionParts As String() = Version.Split(CType(".", Char()))
+            PermissionSet.RevertAssert()
+
+            'Re: version parts, we have the major, minor, build, revision. We key off major+minor.
+            Return Guid.ToString + VersionParts(0) + "." + VersionParts(1)
+        End Function
+
+        Private Function GetTypeLibGuidForAssembly(_assembly As Assembly) As Guid
+            Dim CustomAttributes As Object() = _assembly.GetCustomAttributes(GetType(GuidAttribute), True)
+            If CustomAttributes.Any Then
+                Dim attribute As GuidAttribute = CType(CustomAttributes(0), GuidAttribute)
+                Return New Guid(attribute.Value)
+            End If
+            Return Nothing
+        End Function
+
+        ''' <summary>
+        '''     Uses a named pipe to send the currently parsed options to an already running instance.
+        ''' </summary>
+        ''' <param name="namedPipePayload"></param>
+        Private Sub NamedPipeClientSendOptions(namedPipePayload As NamedPipeXMLData)
+            Try
+                Using _namedPipeClientStream As New NamedPipeClientStream(".", _namedPipeID, PipeDirection.Out)
+                    _namedPipeClientStream.Connect(3000)
+                    ' Maximum wait 3 seconds
+
+                    Dim _xmlSerializer As New XmlSerializer(GetType(NamedPipeXMLData))
+                    _xmlSerializer.Serialize(_namedPipeClientStream, namedPipePayload)
+                End Using
+            Catch ex As Exception
+                ' Error connecting or sending
+            End Try
```

Why swallow exceptions rather than allow exceptions to reach the caller of `Run()`? How will the application recognize failures?

---

The only error here is that the first instance ended between when you thought you were not the first instance and you tried to send the message. This function returns, the caller releases the semaphore and exits. If it throws the caller will need to catch it anyway and exit.

---

Is the exception that is raised in this known failure case known and consistent?